dsgrid/config/project_config.py

```diff
@@ -460,15 +470,16 @@ def load_dimensions(self, dimension_manager: DimensionRegistryManager):
         supplemental_dimensions = dimension_manager.load_dimensions(
             self.model.dimensions.supplemental_dimension_references
         )
-        dims = list(itertools.chain(base_dimensions.values(), supplemental_dimensions.values()))
+        self._base_dimensions.update(base_dimensions)
+        self._supplemental_dimensions.update(supplemental_dimensions)
+
+        dims = list(self.iter_dimensions())
         check_uniqueness((x.model.name for x in dims), "dimension name")
+        check_uniqueness((x.model.query_name for x in dims), "dimension query name")
```

Comment:
This will need to change if we are able to ignore uniqueness for trivial dimensions

Reply:
Another way to deal with my issue would be to only force uniqueness within a DimensionType, but I'm not sure if that would work with the way the queries are set up.